internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -48,26 +49,25 @@ func (m *Metal) Configuration(ctx context.Context) ([]byte, error) {
 
 	u, err := url.Parse(*option)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to parse talos.config url: %w", err)
 	}
 
 	values := u.Query()
 
 	if len(values) > 0 {
-		for key := range values {
+		for key, qValues := range values {
 			switch key {
 			case "uuid":
-				s, err := smbios.New()
-				if err != nil {
-					return nil, err
-				}
-
-				uuid, err := s.SystemInformation().UUID()
-				if err != nil {
-					return nil, err
+				if len(qValues) != 1 {
```

Comment:
I'm not sure I understand why we need to check the length here.

Reply:
In case the url param has no value, e.g. `?uuid=` then `qvalues` will be 0. If this is the case, then we shall fetch the uuid. In case the URL param is `?uuid=<value>` then `qvalues` will be 1. In case the `uuid` param is repeated, instead of using the first one, we just fetch.

Reply:
Ah. Got it. I guess I am so caught up on why the lookup fails. This works, but I'd rather figure out why the uuid lookup fails. That being said I won't block this.

Reply:
Yea it's hard to debug without being able to attach a remote debugger tho. That might be an interesting project, being able to attach a remote debugger if `debug` is enabled.